### PR TITLE
Simplify check_cycle() implement

### PR DESCRIPTION
**Pull request by chenglongyan**
DAG.topological_sort() seems did the same thing?

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/utils/dag_cycle_tester.py [hunk 1]
```diff
@@ -49,33 +42,4 @@ def check_cycle(dag: 'DAG') -> None:
...
-            if visited[current_task_id] == CYCLE_NEW:
-                visited[current_task_id] = CYCLE_IN_PROGRESS
-            task = task_dict[current_task_id]
-            child_to_check = _check_adjacent_tasks(current_task_id, task)
-            if not child_to_check:
-                visited[current_task_id] = CYCLE_DONE
-                path_stack.pop()
-            else:
-                path_stack.append(child_to_check)
+    dag.topological_sort(include_subdag_tasks=False)
```
Comment (mik-laj) [inline airflow/utils/dag_cycle_tester.py:45]: Is the sort implementation recursive or not?

Comment (uranusjr) [reply airflow/utils/dag_cycle_tester.py:45]: Would be nice if we could add a few tests for this to find out.

Comment (uranusjr) [reply airflow/utils/dag_cycle_tester.py:45]: `topological_sort` calls `topological_sort` on each sub-DAG, but with `include_subdag_tasks` (which excludes sub-DAGs), it is _not_ recursive.

**Comment by uranusjr** [comment]
If the two functions are indeed equivalent, perhaps we should deprecate `check_cycle` altogether. Also if we are to replace `check_cycle` with `topological_sort` (whether to deprecate the former or not), the latter needs to be modified to raise `AirflowDagCycleException` instead, for backward compatibility. (Fortunately it currently raise `AirflowException`, which is the superclass of `AirflowDagCycleException`, so changing the exception type would be backward compatible.)

File: airflow/utils/dag_cycle_tester.py (hunk 1 above)
Comment (chenglongyan) [reply airflow/utils/dag_cycle_tester.py:45]: yes , if include_subdag_tasks==True and DAG has sub-DAG , it would be recursived.

**Comment by chenglongyan** [comment]
I think check_cycle should be kept, because it has more readability.

**Pull request closed without merging**